lib/util.js

```diff
@@ -14,23 +14,62 @@ const iconv = require('iconv');
 const USER_AGENT = 'libingester';
 
 const requestDefaults = {
-    pool: { maxSockets: 10 },
     headers: {
         'User-Agent': USER_AGENT,
     },
+    pool: {
+        maxSockets: 10,
+    },
 };
+
 let pooledRequest = request.defaults(requestDefaults);
 
+// Exponential wait: 800ms, 1600ms, 3200ms
+const MAX_RETRIES = 3;
+const RETRY_BACKOFF_DELAY = 800;
+
 // Wrap request results in a promise
-function rp (options) {
+function promisifiedRequest (options) {
     return new Promise((resolve, reject) => {
         pooledRequest(options, (err, res, body) => {
-            if (err) return reject(err);
+            if (err) {
+               return reject(err);
+            }
+
             resolve(res);
         });
     });
 }
 
+function delayExecution (timeout) {
+    return new Promise(resolve => {
+        setTimeout(resolve, timeout);
+    });
+}
+
+function fetchWithRetry (options, attempt = 1) {
+    return promisifiedRequest(options).then(result => {
+        return result;
+    })
+    .catch(err => {
+        if (attempt > MAX_RETRIES) {
+            console.warn(`WARN: Max retries (${MAX_RETRIES}) exceeded!`);
+            throw err;
+        }
+
+        if (err.code == 'ECONNRESET' || err.code == 'ETIMEDOUT') {
+            const timeout = Math.pow(2, attempt) * RETRY_BACKOFF_DELAY;
+            console.warn(`Delaying execution of ${options.uri} by ${timeout}`);
+            return delayExecution(timeout).then(() => {
+                return fetchWithRetry(options, attempt + 1);
+            });
+        }
+
+        // We reach here if unexpected error or we go over MAX_RETRIES
```

Comment:
nit: comment isn't quite correct, this only happens if it's an unexpected error -- exceeding retries happens above